src/batou_ext/nix.py

```diff
@@ -630,7 +630,7 @@ def component_to_nix(component: Component):
             pass
         elif inspect.ismethod(value) or inspect.isgenerator(value):
             pass
-        elif name in ("sub_components", "changed"):
+        elif name in ("sub_components", "changed", "instances"):
```

Comment:
Doesn't that prevent any attribute `instances` from being dumped?

Reply:
While that is true, the attribute `instances` is part of the `Component` class: https://github.com/flyingcircusio/batou/blob/52f9b343e889e1cef53c690873d55ba37e28749e/src/batou/component.py#L155).

I am not sure how to deal with Components that overwrite the instances attribute here, there is probably no easy way to tell them apart.

Reply:
Renaming the attribute to something more obvioulsy internal like `component_instances` or just `_instances` would probably be the best way forward but is outside the scope of this pr.

Reply:
Oh gosh … `instances` is indeed some bad name. I really imagine deployments using it. cc @elikoga